### PR TITLE
Support break for websockets

### LINKED ISSUE #373: Show websockets messages in new tab
Similar (and next) to the history tab and as per the desktop Websockets tab.
I'm thinking that this should only show websockets messages to the current site being displayed? Or is that an option we give to users?
This work can be phased:

- [x] New tab showing websockets messages
- [x] Select message to display and resend
- [x] Option to resend message
- [ ] Support break with websockets

This will require some changes to the ZAP websockets add-on

### PR DESCRIPTION
**Pull request by psiinon**
Fixes #373
Depends on https://github.com/zaproxy/zap-extensions/pull/1956

Need to add a websockets tutorial page.
Only problem is that it reports: [Vue warn]: Avoid mutating a prop directly since the value will be overwritten whenever the parent component re-renders. Instead, use a data or computed property based on the prop's value. Prop being mutated: "payload"

Cant see why its failing for this and not the HTTP one - anyone have any ideas?

### CODE REVIEW

**Comment by psiinon** [comment]
Updated to match the latest changes pushed to https://github.com/zaproxy/zap-extensions/pull/1956

**Comment by psiinon** [comment]
Just added a WebSockets tutorial page, without a task for now.

**Comment by psiinon** [comment]
Its worth noting that websockets break currently doesnt work in daemon mode. I'm looking at that, but its going to be a ZAP issue rather than a HUD one.

**Comment by psiinon** [comment]
@dscrobonia any suggestions re the vuejs error messages?

**Comment by dscrobonia** [comment]
Hey @psiinon When I run this I don't get that Vue.js error. Are you still getting it?

**Comment by dscrobonia** [comment]
I also don't get get any websocket messages intercepted or any indication that websocket breaking is working though. :/ I'm assuming your changes to ZAP Desktop made it to the weekly build by now so that running ./gradlew runZap would be sufficient?

**Comment by thc202** [comment]
Depends which weekly you have, you might need to get a newer one (i.e. remove the directory `build/testZapInstall`).

**Comment by psiinon** [comment]
If other people arent getting the error should we merge this?

**Comment by dscrobonia** [comment]
> Depends which weekly you have, you might need to get a newer one (i.e. remove the directory build/testZapInstall).

@thc202 do you think we should remove this directory when running the `./gradlew clean` command?

**Comment by thc202** [comment]
It does already, it removes everything in the build directory. I suggested removing just that one to keep the other build files (which can be reused in following builds).

**Comment by psiinon** [comment]
@dscrobonia is this working for you yet?
Be really good to get it in ZAP 2.8.0...

**Comment by dscrobonia** [comment]
Yeah looks good to me! Let's get it in!

File: src/main/zapHomeFiles/hud/display.js [hunk 1]
```diff
@@ -678,6 +757,8 @@ document.addEventListener("DOMContentLoaded", () => {
 			historyMessageModalTitle: I18n.t("history_http_message_title"),
 			isWebsocketMessageModalShown: false,
 			websocketMessageModalTitle: I18n.t("websockets_message_title"),
+			isBreakWebSocketMessageModalShown: false,
+			breakWebSocketMessageModalTitle: I18n.t("break_http_message_title"),
```
Comment (dscrobonia) [inline src/main/zapHomeFiles/hud/display.js]: Oh one small change here! This should be 
```
breakWebSocketMessageModalTitle: I18n.t("break_intercpet_ws_title"),
```

Comment (dscrobonia) [reply src/main/zapHomeFiles/hud/display.js]: Its showing the wrong title on intecept. after that, lets merge it!

**Pull request merged**